### PR TITLE
[FSK-97]: Auth Integrate Login Flow With The be

### PR DESCRIPTION
**Pull request by erfan-goodarzi**
This PR creates login integration with the backend API.

## Changes
* Always use `Navigate`, `useNavigate` from `@camp/router`
* All routes changed to: -> `/auth/*` and `/dashboard/*`
* `Route.path` type is now type-safe.
* `GuardLayout` removed in favor of `AuthGuard` and `GuestGuard`.
* `NavLink.path` is type-sfae.
* Added `FullPageLoader` component.
* Move state update to `@camp/data-layer`
* Renamed `isAuth` to `AuthState`.

### CODE REVIEW

File: cypress/e2e/authorization.cy.ts [hunk 1]
```diff
@@ -8,9 +8,9 @@ describe('Authorization', () => {
     cy.location('pathname').should('eq', '/login');
   });
 
-  // NOTE this is in scope of FSK-97
-  it.skip('should be redirected to the families page after successful login', () => {
+  it('should be redirected to the families page after successful login', () => {
     cy.login();
+    cy.visit('/families');
```
Comment (ASafaeirad) [inline cypress/e2e/authorization.cy.ts:13]: You should not visit the families page, login should redirect you to the families page.

File: cypress/e2e/authorization.cy.ts [hunk 2]
```diff
@@ -22,7 +22,6 @@ describe('Authorization', () => {
 
   it('should not be able to go to login after logging in', () => {
     cy.login();
-    cy.visit('/login');
```
Comment (ASafaeirad) [inline cypress/e2e/authorization.cy.ts:25]: You must visit the login page, but it should not let you.

File: src/data-layer/operations/queries/AdminList.query.graphql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# query getAdminList {
```
Comment (ASafaeirad) [inline src/data-layer/operations/queries/AdminList.query.graphql:1]: Delete this file

File: src/data-layer/variables/isAuth.tsx [hunk 4]
```diff
@@ -1,5 +1,18 @@
...
 
 const clientCookie = getClientCookie();
 export const isAuthVar = makeVar(clientCookie.isAuth);
+
+export const loginLocally = () => {
+  const getCookie = getClientCookie();
+
+  if (!getCookie.isAuth) {
+    isAuthVar(true);
+  }
```
Comment (ASafaeirad) [inline src/data-layer/variables/isAuth.tsx:12]: Why set auth to true When `isAuth` is false? :thinking:

File: src/domain/ServerErrors.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  if (!(error instanceof Error))
+    throw new Error('the arg type should be an error');
+
+  switch (error.message) {
+    case invalidCredentials:
+      return messages.errors.INVALID_CREDENTIALS;
+    case internalServer:
+      return messages.errors.INTERNAL_SERVER;
+    default:
+      throw new Error('error');
```
Comment (ASafaeirad) [inline src/domain/ServerErrors.ts:16]: Maybe unknown error

File: src/pages/Login/LoginForm.tsx [hunk 6]
```diff
@@ -77,9 +89,30 @@ export const LoginForm = () => {
...
+            title={errMsg}
+            color="red"
+            sx={theme => ({
+              'color': theme.colors.errorDefault[6],
+              'padding': '12px 14px',
+              '.mantine-rtl-Alert-title': {
+                fontSize: 12,
+                fontWeight: 400,
+              },
+            })}
```
Comment (ASafaeirad) [inline src/pages/Login/LoginForm.tsx:111]: This style does not belong here.

File: src/pages/Login/LoginForm.tsx [hunk 7]
```diff
@@ -77,9 +89,30 @@ export const LoginForm = () => {
...
+          >
+            {''}
```
Comment (ASafaeirad) [inline src/pages/Login/LoginForm.tsx:113]: This one too

File: src/pages/Login/LoginForm.tsx [hunk 8]
```diff
@@ -51,8 +50,21 @@ export const LoginForm = () => {
...
+  const onSubmit = handleSubmit(async ({ username, password }: FormInputs) => {
+    try {
+      await login({
+        variables: { input: { username, password } },
+      });
+      loginLocally();
+      navigate({ to: '/families', replace: true });
+    } catch (err) {
+      const clientError = toClientErrorMessage(err);
+      setErrMsg(clientError);
```
Comment (ASafaeirad) [inline src/pages/Login/LoginForm.tsx:62]: We should unset errors somewhere, also.

**Review by ASafaeirad — CHANGES_REQUESTED** [review]

File: src/data-layer/variables/isAuth.tsx (hunk 4 above)
Comment (erfan-goodarzi) [reply src/data-layer/variables/isAuth.tsx:12]: I added it temporary. just for completing the login flow . if i'm not mistaken the problem is about that from backend side there is no cookie.

File: src/domain/ServerErrors.ts (hunk 5 above)
Comment (erfan-goodarzi) [reply src/domain/ServerErrors.ts:16]: The message? okay.

File: src/pages/Login/LoginForm.tsx (hunk 7 above)
Comment (erfan-goodarzi) [reply src/pages/Login/LoginForm.tsx:113]: if u mean this `{''}` The children of  `Alert` component is required. but based on design we have to use title for our error messages.

File: src/pages/Login/LoginForm.tsx (hunk 8 above)
Comment (erfan-goodarzi) [reply src/pages/Login/LoginForm.tsx:62]: Can u explain more?

**Review by ASafaeirad — APPROVED** [review]

**Pull request merged**